File: Java/app/src/main/java/com/tencent/tcmpp/demo/activity/BaseActivity.java
package com.tencent.tcmpp.demo.activity;

import android.content.Context;
import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.core.view.WindowInsetsControllerCompat;

import com.tencent.tcmpp.demo.utils.LocalUtil;
import com.tencent.tcmpp.demo.utils.LocaleContextWrapper;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(LocaleContextWrapper.create(newBase, LocalUtil.current()));
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void setContentView(int layoutResID) {
        super.setContentView(layoutResID);
        setupSystemBars();
    }

    @Override
    public void setContentView(View view) {
        super.setContentView(view);
        setupSystemBars();
    }

    /**
     * Setup system bars with edge-to-edge display and transparent navigation bar
     */
    private void setupSystemBars() {
        Window window = getWindow();
        
        androidx.core.view.WindowCompat.setDecorFitsSystemWindows(window, false);
        
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS | 
                             WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
            window.setStatusBarColor(Color.TRANSPARENT);
            window.setNavigationBarColor(Color.TRANSPARENT);
        }
        
        WindowInsetsControllerCompat controllerCompat = ViewCompat.getWindowInsetsController(window.getDecorView());
        if (controllerCompat != null) {
            controllerCompat.show(WindowInsetsCompat.Type.statusBars());
            controllerCompat.show(WindowInsetsCompat.Type.navigationBars());
            controllerCompat.setAppearanceLightNavigationBars(true);
            controllerCompat.setAppearanceLightStatusBars(true);
        }
        
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
            window.setStatusBarContrastEnforced(false);
            window.setNavigationBarContrastEnforced(false);
        }
        
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
            window.setNavigationBarDividerColor(Color.TRANSPARENT);
        }
        
        setupContentInsets();
    }

    /**
     * Setup content area insets handling
     * No manual padding needed as layout uses fitsSystemWindows="true"
     */
    private void setupContentInsets() {
        // System automatically handles content spacing with system bars
    }
}

File: Kotlin/app/src/main/java/com/tencent/tcmpp/demo/activity/BaseActivity.kt
package com.tencent.tcmpp.demo.activity

import android.content.Context
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.view.View
import android.view.Window
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat
import com.tencent.tcmpp.demo.utils.LocalUtil
import com.tencent.tcmpp.demo.utils.LocaleContextWrapper

abstract class BaseActivity : AppCompatActivity() {

    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(LocaleContextWrapper.create(newBase, LocalUtil.current()))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun setContentView(layoutResID: Int) {
        super.setContentView(layoutResID)
        setupSystemBars()
    }

    override fun setContentView(view: View) {
        super.setContentView(view)
        setupSystemBars()
    }

    /**
     * Setup system bars with edge-to-edge display and transparent navigation bar
     */
    private fun setupSystemBars() {
        val window: Window = window
        
        androidx.core.view.WindowCompat.setDecorFitsSystemWindows(window, false)
        
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            window.clearFlags(
                WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS or 
                WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION
            )
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
            window.statusBarColor = Color.TRANSPARENT
            window.navigationBarColor = Color.TRANSPARENT
        }
        
        val controllerCompat = ViewCompat.getWindowInsetsController(window.decorView)
        controllerCompat?.let {
            it.show(WindowInsetsCompat.Type.statusBars())
            it.show(WindowInsetsCompat.Type.navigationBars())
            it.isAppearanceLightNavigationBars = true
            it.isAppearanceLightStatusBars = true
        }
        
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
            window.isStatusBarContrastEnforced = false
            window.isNavigationBarContrastEnforced = false
        }
        
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
            window.navigationBarDividerColor = Color.TRANSPARENT
        }
        
        setupContentInsets()
    }

    /**
     * Setup content area insets handling
     * No manual padding needed as layout uses fitsSystemWindows="true"
     */
    private fun setupContentInsets() {
        // System automatically handles content spacing with system bars
    }
}